content/pt/docs/concepts/containers/container-lifecycle-hooks.md

```diff
@@ -0,0 +1,114 @@
+
+Os contêineres podem acessar um _hook_ implementando e registrando um manipulador para esse _hook_.
+Existem dois tipos de manipuladores de _hooks_ que podem ser implementados para contêineres:
+
+* Exec - Executa um comando específico, como `pre-stop.sh`, dentro do cgroups e Namespaces do contêiner.
+* HTTP - Executa uma requisição HTTP em um endpoint específico do contêiner.
+
+### Execução do manipulador de hook
+
+
+Quando um _hook_ de gerenciamento de ciclo de vida do contêiner é chamado, o sistema de gerenciamento do Kubernetes executa o manipulador de acordo com a ação do _hook_, `httpGet` e `tcpSocker` são executados pelo processo kubelet e `exec` é executado pelo contêiner.
+
+As chamadas do manipulador do _hook_ são síncronas no contexto do Pod que contém o contêiner.
+Isso significa que para um _hook_ `PostStart`, o ENTRYPOINT do contêiner e o _hook_ disparam de forma assíncrona.
+No entanto, se o _hook_ demorar muito para ser executado ou travar, o contêiner não consegue atingir o estado `running`.
+
+
+Os _hooks_ `PreStop` não são executados de forma assíncrona a partir do sinal para parar o conêiner, o _hook_ precisa finalizar a sua execução antes que o sinal TERM possa ser enviado.
+Se um _hook_ `PreStop` travar durante a execução, a fase do Pod será `Terminating` e permanecerá até que o Pod seja morto após seu `terminationGracePeriodSeconds` expirar. Esse período de tolerância se aplica ao tempo total necessário
+para o _hook_ `PreStop`executar e para o contêiner parar normalmente.
+Se por exemplo, o `terminationGracePeriodSeconds` é 60, e o _hook_ leva 55 segundos para ser concluído, e o contêiner leva 10 segundos para parar normalmente após receber o sinal, então o contêiner será morto antes que possa parar
+normalmente, uma vez que o `terminationGracePeriodSeconds` é menor que o tempo total (55 + 10) que é necessário para que essas duas coisas aconteçam.
+
+Se um _hook_ `PostStart` ou `PreStop` falhar, ele mata o contêiner. 
+
+Os usuários devem tornar seus _hooks_ o mais leve possíveis.
+Há casos, no entanto, em que comandos de longa duração fazem sentido, como ao salvar o estado
+antes de parar um contêiner.
+
+### Garantias de entrega de _hooks_
+
+A entrega do _hook_ é destinada a acontecer *pelo menos uma vez*,
+o que quer dizer que um _hook_ pode ser chamado várias vezes para qualquer evento, 
+como para `PostStart` ou `PreStop`.
+Depende da implementação do _hook_ lidar com isso corretamente.
+
+Geralmente, apenas entregas únicas são feitas.
+Se, por exemplo, um receptor de _hook_ HTTP estiver inativo e não puder receber tráfego,
+não há tentativa de reenviar.
+Em alguns casos raros, no entanto, pode ocorrer uma entrega dupla.
+Por exemplo, se um kubelet reiniciar no meio do envio de um _hook_, o _hook_ pode ser 
+reenviado depois que o kubelet voltar a funcionar.
+
+### Depurando manipuladores de _hooks_
+
+Os logs para um manipulador de _hook_ não expostos em eventos de Pod.
```

Comment:
I guess the correct here would be 'são expostos em eventos de Pod"

Reply:
The original is "The logs for a Hook handler are not exposed in Pod events.".  I added the missing word, thanks!